### ISSUE TITLE
Unlocked keychain is never locked

### ISSUE THREAD

**Issue by ivany4** [carlossless/tangerine#5]
Correct me if I'm wrong, but reset_keychain() simply resets default keychain to login, but does not lock previously unlocked keychain in $KEYCHAIN_PATH.

**Comment by carlossless** [comment]
Yep you're actually right. But now that I think about it, I don't know if locking the keychain makes sense. Probably from a security standpoint it does and we could check the state of the key chain prior to unlocking, save it, and then after building the app lock the keychain if it was locked prior to launching.

What I don't like much about this is that given parallel builds (which we can do if the simulator isn't being used), this would probably fail as one process would lock the keychain in the middle of the other processes work, causing the build to fail.

**Comment by carlossless** [comment]
This whole file-state based approach is really meh. I wish Apple had implemented this feature through some kind of session mechanism, where you would unlock the keychain with a per-process session.
